File: todo-reservation-springboot/src/main/java/de/vkb/dojo/es/reservations/kafka/config/StoreNames.java
package de.vkb.dojo.es.reservations.kafka.config;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Configuration;

@Configuration
public class StoreNames {
    @Value("${spring.kafka.streams.reservationCommandHandler.stores.rooms}")
    public String reservationCommandRooms;
    @Value("${spring.kafka.streams.reservationCommandHandler.stores.persons}")
    public String reservationCommandPersons;
    @Value("${spring.kafka.streams.reservationEventAggregator.stores.aggregate}")
    public String reservationEventAggregate;
    @Value("${spring.kafka.streams.reservationRoomSaga.stores.rooms}")
    public String reservationRoomSagaRooms;
    @Value("${spring.kafka.streams.reservationPersonSaga.stores.persons}")
    public String reservationPersonSagaPersons;
    @Value("${spring.kafka.streams.reservationReader.stores.state}")
    public String reservationReaderState;
    @Value("${spring.kafka.streams.feedbackReader.stores.state}")
    public String feedbackReaderState;
}

File: todo-reservation-micronaut/src/main/kotlin/de/vkb/dojo/es/reservations/kafka/config/StoreNames.kt
package de.vkb.dojo.es.reservations.kafka.config

import io.micronaut.context.annotation.ConfigurationProperties
import io.micronaut.context.annotation.Property
import io.micronaut.context.annotation.Value
import jakarta.inject.Singleton

@Singleton
data class StoreNames(
    @Property(name = "kafka.streams.reservation-command-handler.stores.rooms") val reservationCommandRooms: String,
    @Property(name = "kafka.streams.reservation-command-handler.stores.persons") val reservationCommandPersons: String,
    @Property(name = "kafka.streams.default.stores.aggregate") val reservationEventAggregate: String,
    @Property(name = "kafka.streams.reservation-room-saga.stores.rooms") val roomSagaStore: String,
    @Property(name = "kafka.streams.reservation-person-saga.stores.persons") val personSagaStore: String,
    @Property(name = "kafka.streams.reservation-reader.stores.state") val reservationReaderState: String,
    @Property(name = "kafka.streams.reservation-feedback-reader.stores.state") val feedbackReaderState: String
)
